### PR TITLE
Add Timer tests and fix progress

### PR DESCRIPTION
**Pull request by cmc5788**
# Description

#610 fixed a bug with `Timer`; this PR adds some tests for this and other `Timer` functions.

Adding these tests exposed a bug with `progress`, which is also fixed in this PR.

Screen Shot 2021-01-05 at 7 22 11 PM

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

If something is unclear, please submit the PR anyways and ask about what you thought was unclear.

- [x] This branch is based on the latest `master`
- [x] I have added an entry under `[next]` in `CHANGELOG.md`
- [x] I have formatted my code with `flutter format`
- [ ] I have made corresponding changes to the documentation
- [ ] I have added examples for new features in `doc/examples`
- [x] The continuous integration (CI) is passing

### CODE REVIEW

**Review by spydon — APPROVED** [review]
Great!

**Review by erickzanardo — APPROVED** [review]

**Review by wolfenrain — APPROVED** [review]

**Comment by cmc5788** [comment]
Thanks all :) as before, if there aren't any big issues feel free to merge whenever you'd like since I don't have write access here.

**Comment by erickzanardo** [comment]
> Thanks all :) as before, if there aren't any big issues feel free to merge whenever you'd like since I don't have write access here.

Sure thing, we are just waiting for at least one more approval :)

**Pull request merged**
